persistence-elasticsearch/core/src/main/java/org/apache/unomi/persistence/elasticsearch/ElasticSearchPersistenceServiceImpl.java

```diff
@@ -1346,13 +1349,22 @@ private long queryCount(final QueryBuilder filter, final String itemType) {
 
             @Override
             protected Long execute(Object... args) throws IOException {
+
+                CountRequest countRequest = new CountRequest(getIndexNameForQuery(itemType));
+                SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+                searchSourceBuilder.query(filter);
+                countRequest.source(searchSourceBuilder);
+                CountResponse response = client.count(countRequest, RequestOptions.DEFAULT);
+                return response.getCount();
+                /*
                 SearchRequest searchRequest = new SearchRequest(getIndexNameForQuery(itemType));
                 SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
                 searchSourceBuilder.query(filter);
                 searchSourceBuilder.size(0);
                 searchRequest.source(searchSourceBuilder);
                 SearchResponse response = client.search(searchRequest, RequestOptions.DEFAULT);
                 return response.getHits().getTotalHits().value;
+                 */
```

Comment:
Maybe, we should drop this  code in the block comment.

Reply:
Indeed you are right

Reply:
I've removed it